### PR TITLE
Test coverage

### PR DESCRIPTION
**Pull request by cpotteck**
Work on improving test coverage:
* adding all python files to some test suite
* using the coverage report to cover missing lines
* compiling important test scenarios and edge cases that should be covered.

I will keep this rebased on master until merge, so that it stays up-to-date.

### CODE REVIEW

File: functions/common.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def factorial(argument: int) -> int:
         result *= i
     return result
 
-
+# Note - I don't think this is actually used anywhere
```
Comment (EmanuelUnity3D) [inline functions/common.py:31]: It isn't. I removed it at one point but it got added back so I added testing for it in my last PR. I remember having issues with it actually but I can't remember what they were anymore.

File: tests/test_calculate.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        self.assertEqual(calculate.evaluate('0^(-1)', False, False), 'Input Error: Inverse of 0 is undefined.')
+        self.assertEqual(calculate.evaluate('sqrt(-2)', False, False), 'Input Error: Not in domain of radical function.')
+        self.assertEqual(calculate.evaluate('(-5)^2.1', False, False), 'Input Error: Negative base with fractional exponent.')
+        self.assertEqual(calculate.evaluate('5', False, True), 'Input Error: Not in binary.')
+        self.assertEqual(calculate.evaluate('1.5', False, True), 'Input Error: Not in binary.')
+
+
+    def test_arithmetic_error(self):
+        self.assertEqual(calculate.evaluate('1/0', False, False), 'Arithmetic Error: Division by 0 is undefined.')
+        self.assertEqual(calculate.evaluate('5%0', False, False), 'Arithmetic Error: Division by 0 is undefined.')
```
Comment (EmanuelUnity3D) [inline tests/test_calculate.py:43]: I actually made this exception an input error because I forgot that we were using arithmetic exceptions for divisions by 0. Does this test still pass?

Comment (EmanuelUnity3D) [reply tests/test_calculate.py:43]: Oh wait, never mind. We're not using our modulo function in the actual calculator.

File: tests/test_common.py [hunk 3]
```diff
@@ -17,11 +17,14 @@ def test_factorial(self):
         with self.assertRaises(the_exception.InputError):
             common.factorial(56.3)
 
+    # should be removed if our mod function is not in use
+    # some of these 6%3 and 3%6 both fail
```
Comment (EmanuelUnity3D) [inline tests/test_common.py:21]: Okay yeah these are the issues that I was talking about. I didn't know how to fix them at the time. I'm okay with removing the tests but we should remove the mod function too then. I added the tests because I didn't want to leave part of the code uncovered and include something broken. If we get rid of the tests without fixing the issue then that defeats the purpose of them.

Comment (cpotteck) [reply tests/test_common.py:21]: yep, i will remove the function + tests. Just wanted to confirm.

**Comment by EmanuelUnity3D** [comment]
Looks pretty good. Some tests remain to be added but everything added so far looks good.

File: tests/test_common.py (hunk 3 above)
Comment (EmanuelUnity3D) [reply tests/test_common.py:21]: Sounds good.

File: memory.py [hunk 4]
```diff
@@ -1,8 +1,9 @@
 import sys
 import re
-from nparser import Parser
+from parse import Parser
+
+expression = sys.argv[1] if len(sys.argv) > 1 else ""
```
Comment (VisusAdAstra) [inline memory.py:5]: ??? why we need this condition

File: tests/test_calculate.py (hunk 2 above)
Comment (VisusAdAstra) [reply tests/test_calculate.py:43]: do we need number of precision for assertAlmostEqual (7 or 9)

File: memory.py (hunk 4 above)
Comment (cpotteck) [reply memory.py:5]: testing breaks if this condition is not there bc theres no arguments
